app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Katello
+  class Api::Rhsm::CandlepinDynflowProxyController < Api::V2::ApiController
+    include Katello::Authentication::ClientAuthentication
+    include ForemanTasks::Triggers
+    include AbstractController::Callbacks
+
+    before_action :find_host, :only => [:upload_package_profile]
+    before_action :authorize_client_or_user, :only => [:upload_package_profile]
+
+    #api :PUT, "/consumers/:id/packages", N_("Update installed packages")
+    #api :PUT, "/consumers/:id/profile", N_("Update installed packages")
+    #param :id, String, :desc => N_("UUID of the consumer"), :required => true
+    def upload_package_profile
+      User.as_anonymous_admin do
+        async_task(::Actions::Katello::Host::UploadPackageProfileWrapper, @host, request.body.string)
+      end
+      [
+        200,
+        { "Content-Type" => "application/json" },
+        Resources::Candlepin::Consumer.get(@host.subscription_facet.uuid)
+      ]
+    end
```

Comment:
you can easily stub the controller .async_task action, we do that in a number of places such as: 

https://github.com/Katello/katello/blob/master/test/controllers/api/v2/content_view_versions_controller_test.rb#L123

so you could test before and after the action with the test stub and count the allocation delta, that would be a good test!